js/app.js

```diff
@@ -1,78 +1,148 @@
+Product.elementObject = document.getElementById('chooseProduct'); //HTML location of form
+Product.elementObject.addEventListener('submit', handleVote); //Create event listener
 
 /**
  * Constructor for listed items.
  * @param {string} url Location of product image in img directory
+ * @param {string} name Unique product identifying name
  */
 function Product(url, name) {
   this.url = url;
   this.name = name;
   this.votes = 0;
   this.displays = 0;
-  allProducts.push(this);
+  Product.allNames.push(this.name);
+  Product.allProducts.push(this);
 }
 
 /**
- * select new image
- * compare image to previous three images
- * store into currentDisplay[]
+ * Generates Random Index Numbers
  */
-function selectImages() {
-  var holdingArray = [];
-  for(var i = 0; i < NUMBER_DISPLAYED; i++) {
+function getIndexNumbers() {
+  for(var i = 0; i < Product.NUMBER_DISPLAYED; i++) {
     do {
-      var randomNum = Math.floor(Math.random() * allProducts.length);
-      holdingArray.push(allProducts[randomNum]);
-    } while (currentDisplay.contains(holdingArray[i]));
+      var randomIndex = Math.floor(Math.random() * Product.allProducts.length);
+    } while (Product.currentDisplay.includes(randomIndex) || Product.indexHolder.includes(randomIndex));
+    Product.indexHolder.push(randomIndex);
   }
-  currentDisplay = holdingArray;
-  firstImage.src = currentDisplay[0].url;
-  secondImage.src = currentDisplay[1].url;
-  thirdImage.src = currentDisplay[2].url;
+  Product.currentDisplay = Product.indexHolder; //Assign validated index values to current display
+  Product.indexHolder = []; //Reset Index Holder
+  return(Product.currentDisplay);
+}
+
+/**
+ * TODO Figure out how to scale this with NUMBER_DISPLAYED
+ */
+function postImages() {
+  getIndexNumbers();
+  Product.allProducts[Product.currentDisplay[0]].displays++;
+  Product.allProducts[Product.currentDisplay[1]].displays++;
+  Product.allProducts[Product.currentDisplay[2]].displays++;
+  Product.firstImage.src = Product.allProducts[Product.currentDisplay[0]].url;
+  Product.firstImage.alt = Product.allProducts[Product.currentDisplay[0]].name;
+  Product.secondImage.src = Product.allProducts[Product.currentDisplay[1]].url;
+  Product.secondImage.alt = Product.allProducts[Product.currentDisplay[1]].name;
+  Product.thirdImage.src = Product.allProducts[Product.currentDisplay[2]].url;
+  Product.thirdImage.alt = Product.allProducts[Product.currentDisplay[2]].name;
 }
 
+/**
+ * Increments iterations and vote counter for selected object
+ * @param {*} event Called upon submit
+ */
 function handleVote(event) {
-  event.preventDefault(); //Prevent page from reloading
-  var choices = document.getElementsByName('imageChoice');
-  var imageChoice;
-  for (var i = 0; i < choices.length; i++) {
-    if(choices[i].checked) {
-      imageChoice = Number(choices[i].value);
-    }
+  event.preventDefault(); //Prevent Page Refresh
```

Comment:
Right now, if I hit submit on the form without picking a product, I get an error in the JS console--use some validation or a conditional here to prevent that!

Reply:
AND, it doesn't register a vote, but it still counts towards the 25 total votes!